master/static/migrations/20240321174101_handle-single-run-experiments-best-trial.tx.up.sql

```diff
@@ -0,0 +1,52 @@
+CREATE OR REPLACE FUNCTION autoupdate_exp_best_trial_single_run_no_validation() RETURNS trigger AS $$
+BEGIN 
+    WITH rc AS (
+        SELECT COUNT(*) AS count
+        FROM runs
+        WHERE experiment_id = NEW.id),
+    br AS (
+        SELECT id, best_validation_id 
+        FROM runs 
+        WHERE 
+            runs.experiment_id = NEW.id 
+            AND
+            best_validation_id IS NULL
+        ORDER BY searcher_metric_value_signed LIMIT 1)
+    UPDATE 
+        experiments 
+    SET 
+        best_trial_id = br.id
+    FROM br, rc
+    WHERE
+        experiments.id = NEW.id
+        AND
+        rc.count = 1;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS autoupdate_exp_best_trial_single_run_no_validation_trigger ON experiments;
+CREATE TRIGGER autoupdate_exp_best_trial_single_run_no_validation_trigger 
+AFTER UPDATE OF state ON experiments
+FOR EACH ROW 
+    WHEN (NEW.state = 'COMPLETED' AND NEW.best_trial_id IS NULL)
+    EXECUTE PROCEDURE autoupdate_exp_best_trial_single_run_no_validation();
```

Comment:
if we do only care about single searcher experiments, it might be simpler to do this in the code when we create the trial rather than as a db trigger

Reply:
That makes a lot of sense to me, assuming custom searcher isn't an issue!